crates/snp_validation/src/lib.rs

```diff
@@ -287,96 +309,133 @@ impl Snp {
+            report.body.rsvd5 == [0; 168],
+            "snp report reserved field 5 was set"
+        );
+
+        ensure!(
+            report.body.policy.rsvd == [0; 5],
+            "snp report policy reserved fields were set"
+        );
+
+        ensure!(
+            report.body.plat_info.rsvd == [0; 7],
+            "snp report platform_info reserved fields were set"
+        );
 
         // Check fields not set by Enarx
-        for value in report.body.author_key_digest {
-            if value != 0 {
-                return Err(anyhow!(
-                    "snp report author_key_digest field not set by Enarx"
-                ));
+        ensure!(
+            report.body.host_data == [0; 32],
+            "snp report host_data field should not set by Enarx"
+        );
+
+        ensure!(report.body.vmpl == 0, "snp report vmpl field invalid value");
+
+        if let Some(config) = config {
+            ensure!(
+                config.abi.matches(&report.body.policy.into()),
+                "snp minimum abi not met"
+            );
+
+            if !config.measurements.signer.is_empty() {
+                let signed = config
+                    .measurements
+                    .signer
+                    .contains(&validation_common::Digest(report.body.author_key_digest));
+                ensure!(signed, "snp untrusted enarx author_key_digest");
             }
-        }
 
-        for value in report.body.host_data {
-            if value != 0 {
-                return Err(anyhow!("snp report host_data field not set by Enarx"));
+            if !config.id_key_digest.is_empty() {
+                let signed = config
+                    .id_key_digest
+                    .contains(&validation_common::Digest(report.body.id_key_digest));
+                ensure!(
+                    signed,
+                    "snp untrusted enarx id_key_digest not in list of allowed key digests"
+                );
             }
-        }
 
-        for value in report.body.id_key_digest {
-            if value != 0 {
-                return Err(anyhow!("snp report id_key_digest field not set by Enarx"));
+            if !config.id_key_digest_blacklist.is_empty() {
+                let signed = config
+                    .id_key_digest_blacklist
+                    .contains(&validation_common::Digest(report.body.id_key_digest));
+                ensure!(!signed, "snp untrusted enarx id_key_digest in blacklist");
             }
-        }
-
-        if report.body.vmpl != 0 {
-            return Err(anyhow!("snp report vmpl field not set by Enarx"));
-        }
 
-        if report.body.guest_svn != 0 {
-            return Err(anyhow!("snp report guest_svn field not set by Enarx"));
-        }
+            if !config.measurements.hash.is_empty() {
+                let allowed = config
+                    .measurements
+                    .hash
+                    .contains(&validation_common::Digest(report.body.measurement));
```

Comment:
```suggestion
                    .contains(&report.body.measurement);
```

The traits removed by you from the diff where there for a reason.

Reply:
Fixed